components/onboarding_tour/onboarding_tour_manager.tsx

```diff
@@ -246,10 +246,15 @@ const useOnBoardingTourTipManager = (): OnBoardingTourTipManager => {
                 handleNext();
             }
         };
-        window.addEventListener('keydown', handleKeyDown);
-        return () =>
-            window.removeEventListener('keydown', handleKeyDown);
-    }, [handleNext]);
+        if (show) {
+            window.addEventListener('keydown', handleKeyDown);
+        }
+        return () => {
+            if (show) {
+                window.removeEventListener('keydown', handleKeyDown);
+            }
+        };
```

Comment:
Can `show` change from `true` to `false` before the component gets unmounted?
It would be probably best, IMO, if we have the guard inside the `handleKeyDown`.

Reply:
Yeah make sense